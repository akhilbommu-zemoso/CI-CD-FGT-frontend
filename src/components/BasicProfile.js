import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { ProfileFormFields } from './BasicProfileTop'
import Avtar from './Avtar'
import Divider from '@material-ui/core/Divider';
import Snackbar from '@material-ui/core/Snackbar';
import MuiAlert from '@material-ui/lab/Alert';
import FTButton from '../atoms/FTButton'
import ProfileDeleteAccount from './ProfileDeleteAccount'
import ProfileEmail from './ProfileEmail'
import ProfilePassword from './ProfilePassword'
const useStyles = makeStyles((theme) => ({
    padding: {
        padding: "20px"
    },
    colorbutton: {
        marginTop: '50px',
        color: 'white',
        textTransform: 'none',
        marginBottom: '50px',
        backgroundColor: 'rgb(4,136,255)',
        '&:hover': {
            color: 'white',
            backgroundColor: 'rgb(4,136,255)'
        },
    },
    flex: {
        display: "flex",
        justifyContent: "space-between"
    },
    alignRight: {
        width: '100%',
        textAlign: 'right'
    },
    root: {
        width: '100%',
        '& > * + *': {
            marginTop: theme.spacing(2),
        },
    }
}));

export default function BasicProfile(props) {
    const classes = useStyles();
    function Alert(props) {
        return <MuiAlert elevation={6} variant="filled" {...props} />;
      }
      const checkErrorMessage=()=>{
        if(props.errorMessage.name===""&&props.errorMessage.phone==="")
            return false;
        else
            return true;
      }
    return (
        <div>
            <div className={classes.padding}>
                <div className={classes.flex}>
                    <ProfileFormFields
                        id="profileform"
                        values={props.values}
                        handleChange={(event) => props.handleChange(event)}
                        handleChangeCountry={(event)=>props.handleChangeCountry(event)}
                        errorMessage={props.errorMessage}/>
                    <Avtar imageData={props.values.profilePicture}
                        userName = {props.values.name}
                        handleChangePicture={(data)=>props.handleChangePicture(data)}/>
                </div>
            </div>
            <Divider />
            <div className={classes.padding} data-testid="profileemail">
                <ProfileEmail email={props.values.email} />
            </div>
            <Divider />
            <div className={classes.padding} data-testid="profilepassword">
                <ProfilePassword />
            </div>
            <Divider />
            <div className={classes.padding} data-testid="profiledeleteaccount">
                <ProfileDeleteAccount />
            </div>
            <div className={classes.alignRight}>
                    <FTButton
                        data-testid="button"
                        name="Save"
                        method={(event) => props.postToUserApi()}
                        disabled={checkErrorMessage()}
                    />
                <Snackbar
                    data-testid='snackbar'
                    anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
                    open={props.open} autoHideDuration={6000}
                    onClose={(event, reason) => props.handleClose(event, reason)}>
                    <Alert onClose={(event, reason) => props.handleClose(event, reason)} severity="success">
                        Changes Saved!
                    </Alert>
                </Snackbar>
            </div>
        </div>
    );
}