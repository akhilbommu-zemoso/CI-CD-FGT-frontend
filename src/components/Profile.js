import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import ProfileTabs from './ProfileTabs'
import ProfileSideBar from './ProfileSideBar'
import AppBar from './AppBar'
import HomeOutlinedIcon from '@material-ui/icons/HomeOutlined';
import { Redirect } from 'react-router-dom'
import ProfileList from './ProfileList'
import Brand from '../atoms/Brand'
import GenralProfile from '../components/GeneralProfile'
import BasicProfile from '../components/BasicProfile'
import MuiAlert from '@material-ui/lab/Alert';
const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
    paper: {
        padding: theme.spacing(2),
        textAlign: 'center',
        color: theme.palette.text.secondary,
    },
    top: {
        display: 'flex',
        marginTop: '100px',
    },
    margins: {
        marginRight: '10px',
        marginLeft: '10px',
    },
    linksclass: {
        color: 'black',
        paddingTop: '0px',
        textDecoration: 'none',
        '&:hover': {
            color: 'black',
            textDecoration: 'none',
        },
    },
}));
function Profile(props) {
    const [customError, setcustomError] = useState(null);
    const [result, setResult] = useState();
    const [basicData, setBasicData] = useState();
    const [loadingUserData, setLoadingUserData] = useState(false);
    const [loadingSettingData, setLoadingSettingData] = useState(false);
    const [abc, setAbc] = useState();
    const [open, setOpen] = React.useState(false);
    const [errorMessage, setErrorMessage] = useState({ name: "", phone: "" });
    const settingsApi = () => {
        var accessToken = localStorage.getItem("accessToken");
        fetch(process.env.REACT_APP_BACKEND_USER_SETTINGS, {
            method: 'GET',
            headers: {
                'Authorization': 'Bearer ' + accessToken,
            },
        }).then(response => {
            if (!response.ok) {
                throw new Error(response.status)
            }
            return response.json();
        })
            .then((data) => { return data; })
            .then((data) => setResult(data))
            .then(() => { setLoadingSettingData(true) })
            .catch((error) => {
                console.error('Error Occured at:', error, typeof (error));
                if (error.message == "401") {
                    setcustomError("Unauthorized Access");
                }
                else if (error.message == "404") {
                    setcustomError("Invalid User");
                }
                else if (error.message == "500") {
                    setcustomError("Internal Server Error");
                }
                else {
                    setcustomError("Unable to fetch resource")
                }
                return <Redirect to="/error" />
            });
    }
    const userApi = (signal) => {
        var accessToken = localStorage.getItem("accessToken");
        var apiurl = process.env.REACT_APP_BACKEND_USER;

        fetch(apiurl, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': 'Bearer ' + accessToken,
            },
        }).then(response => {
            if (!response.ok) {
                throw new Error(response.status)
            }
            return response.json();
        })
            .then((data) => { return data; })
            .then((data) => setBasicData(data))
            .then(() => setLoadingUserData(true))
            .catch((error) => {
                if (error.message === "401") {
                    setcustomError("Unauthorized Access");
                }
                else if (error.message === "404") {
                    setcustomError("Invalid User");
                }
                else if (error.message === "500") {
                    setcustomError("Internal Server Error");
                }
                else {
                    setcustomError("Unable to fetch resource")
                }
            });

    }
    useEffect(() => {
        userApi();
        settingsApi();
    }, [abc])
    const handleChange = (event) => {
        var res = { ...result };
        var name = event.target.name;
        var array = name.split("_");
        res[array[0]][array[1]] = event.target.checked ? event.target.checked : event.target.value;
        setResult(res);
    };
    const handleChangeCountry = (event) => {
        var countryName = event.target.innerText.split("(")[0].substring(4).trim();
        var name = "country";
        setBasicData({ ...basicData, [name]: countryName });
    }
    const handleChangeUser = (event) => {
        if(event.target.name==="name")
            validateName(event.target.value)
        if(event.target.name==="phone")
            validatePhone(event.target.value)
        setBasicData({ ...basicData, [event.target.name]: event.target.value });
    };
    const validateName = (data) =>{
        let namePattern = /^[a-zA-Z]*[A-Za-z0-9\s]*$/;
        if(namePattern.test(data)){
            setErrorMessage({...errorMessage,"name": ""});
        }
        else
        setErrorMessage({...errorMessage,"name": "User name should have only alphabets"});
    }
    const validatePhone = (data) =>{
        let phoneNumberPattern = /^\+\d{2}\-\d{10}$/;
        if(phoneNumberPattern.test(data)){
            setErrorMessage({...errorMessage,"phone": ""});
        }
        else
        setErrorMessage({...errorMessage,"phone": "Invalid phone number"});
    }
    const handleClose = (event, reason) => {
        if (reason === 'clickaway') {
            return;
        }

        setOpen(false);
    };

    const handleChangePicture = (data) => {
        setBasicData({ ...basicData, ["profilePicture"]: data });
    }
    const postToApi = (event) => {
        var accessToken = localStorage.getItem("accessToken");
        fetch(process.env.REACT_APP_BACKEND_USER_SETTINGS, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': 'Bearer ' + accessToken,
            },
            body: JSON.stringify(result),

        }).then(response => {
            if (!response.ok) {
                console.log("status--------" + response.status);
                throw new Error(response.status)
            }
            return response.json();
        })
            .then((data) => {
                setOpen(true);
            })
            .catch((error) => {
                console.error('Error Occured at:', error, typeof (error));
                if (error.message == "401") {
                    setcustomError("Unauthorized Access");
                }
                else if (error.message == "404") {
                    setcustomError("Invalid User");
                }
                else if (error.message == "500") {
                    setcustomError("Internal Server Error");
                }
                else {
                    setcustomError("Unable to fetch resource")
                }
                return <Redirect to="/error" />
            });
    };
    const postToUserApi = (event) => {
        var accessToken = localStorage.getItem("accessToken");
        fetch(process.env.REACT_APP_BACKEND_USER, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': 'Bearer ' + accessToken,
            },
            body: JSON.stringify(basicData),

        }).then(response => {
            if (!response.ok) {
                console.log("status--------" + response.status);
                throw new Error(response.status)
            }
            return response.json();
        })
            .then((data) => {
                localStorage.setItem("user",JSON.stringify(data));
                console.log("response data",data)
                setBasicData(data);
                updateSettingsUser(data);
             })
            .then(() => {
                setOpen(true);
            })
            .catch((error) => {
                console.error('Error Occured at:', error, typeof (error));
                if (error.message == "401") {
                    setcustomError("Unauthorized Access");
                }
                else if (error.message == "404") {
                    setcustomError("Invalid User");
                }
                else if (error.message == "500") {
                    setcustomError("Internal Server Error");
                }
                else {
                    setcustomError("Unable to fetch resource")
                }
                return <Redirect to="/error" />
            });

    };
    const updateSettingsUser=(data)=>{
        const settings = {...result};
        settings.message.user=data;
        settings.notification.user=data;
        settings.setting.user=data;
        setResult(settings);
    }
    const classes = useStyles();
    return (
        <div>
            {loadingUserData && loadingSettingData &&
                <div>
                    <div>
                        <AppBar brand={<Brand name={"ZE-Transact"} />} profilelist={<ProfileList />}></AppBar>
                    </div>
                    <div className={classes.top}>
                        <Grid item xs={3}>
                            <div className={classes.margins}>
                                <ProfileSideBar />
                            </div>
                        </Grid>
                        <Grid item xs={9}>
                            <div className={classes.margins}>
                                <ProfileTabs
                                    result={result}
                                    method={(event) => handleChange(event)}
                                    handleChangeUser={(event) => handleChangeUser(event)}
                                    postToApi={(event) => postToApi(event)}
                                    postToUserApi={(event) => postToUserApi(event)}
                                    basicData={basicData}
                                    handleChangeCountry={(event) => handleChangeCountry(event)}
                                    handleChangePicture={(data) => handleChangePicture(data)}
                                    open={open}
                                    handleClose={(event, reason) => handleClose(event, reason)}
                                    tab2={
                                        <GenralProfile
                                            result={result}
                                            method={(event) => handleChange(event)}
                                            postToApi={(event) => postToApi(event)}
                                            open={open}
                                            handleClose={(event, reason) => handleClose(event, reason)}
                                        />
                                    }
                                    tab1={
                                        <BasicProfile
                                            values={basicData}
                                            handleChange={(event) => handleChangeUser(event)}
                                            postToUserApi={(event) => postToUserApi(event)}
                                            handleChangeCountry={(event) => handleChangeCountry(event)}
                                            handleChangePicture={(data) => handleChangePicture(data)}
                                            open={open}
                                            handleClose={(event, reason) => handleClose(event, reason)}
                                            errorMessage={errorMessage}
                                        />
                                    }
                                />
                            </div>
                        </Grid>
                    </div>
                </div>

            }
        </div>

    )
}

export default Profile