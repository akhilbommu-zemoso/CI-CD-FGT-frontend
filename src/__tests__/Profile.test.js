import React, { Component } from 'react'
import { render, fireEvent, getAllByText,screen } from '@testing-library/react'
import { renderHook, act } from '@testing-library/react-hooks'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme'
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { create } from "react-test-renderer";
import Profile from '../components/Profile'
import BasicProfile from '../components/BasicProfile';
import fetchMock from 'fetch-mock';
Enzyme.configure({ adapter: new Adapter() });
const data = () => {
    var json = {
        id: "1",
        name: "Fareed Sheik",
        email: "sheikfareed3819@gmail.com",
        auth0Id: "google - oauth2 | 115320575601904541142",
        phone: " + 91 - 7680024324", "country": "United Arab Emirates",
        createdAt: "1589626102282", "createdBy": "Fareed",
        modifiedAt: "1589626102282", "modifiedBy": "Fareed",
        profilePicture: {
            id: "1",
            photoBlob: null,
            photoContentLength: "0",
            photoContentType: null,
        }
    }
    localStorage.setItem('user', JSON.stringify(json))
    return json;
}
it('should take a snapshot', () => {
    data()
    const { asFragment } = render(<Router>
        <Profile values={data()} /></Router>
    )
    expect(asFragment(<Router><Profile values={data()} /></Router>)).toMatchSnapshot()
});

it('should take a snapshot', () => {
    data()
    const { asFragment } = render(<Router>
        <Profile values={data()} /></Router>
    )
    expect(asFragment(<Router><Profile values={data()} /></Router>)).toMatchSnapshot()
});

it("checks api", async () => { 
    fetchMock.mock("http://localhost:8080/api/user/users", {
        body: {
            status: 200,
            data:data()
    }});
    fetchMock.mock("http://localhost:8080/api/usersetting/settings",200)
    localStorage.setItem('accessToken', "abcdfetjjj");
    const { getByText} = render(<Router>
        <Profile values={data()} /></Router>
    )
    const res = await fetch("http://localhost:8080/api/user/users");
    const res1 = await fetch("http://localhost:8080/api/usersetting/settings");
    expect(res.ok).toBeTruthy
    expect(res1.ok).toBeTruthy
   
})







