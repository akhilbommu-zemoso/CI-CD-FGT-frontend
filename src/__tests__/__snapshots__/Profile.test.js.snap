// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`should take a snapshot 2`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
